cardano-api/src/Cardano/Api/Governance/Poll.hs

```diff
@@ -76,29 +90,63 @@ instance HasTypeProxy GovernancePoll where
     data AsType GovernancePoll = AsGovernancePoll
     proxyToAsType _ = AsGovernancePoll
 
+instance AsTxMetadata GovernancePoll where
+    asTxMetadata GovernancePoll{govPollQuestion, govPollAnswers, govPollNonce} =
+      makeTransactionMetadata $ Map.fromList
+        [ ( metadataLabel
+          , TxMetaMap $
+           [ ( TxMetaNumber 0, metaTextChunks govPollQuestion )
+           , ( TxMetaNumber 1, TxMetaList (metaTextChunks <$> govPollAnswers) )
+           ] ++
+           case govPollNonce of
+             Nothing -> []
+             Just nonce -> [ ( TxMetaText "_", TxMetaNumber (toInteger nonce) ) ]
+          )
+        ]
+
 instance SerialiseAsCBOR GovernancePoll where
     serialiseToCBOR =
-      error "not implemented"
-
-    deserialiseFromCBOR AsGovernancePoll _bs =
-      error "not implemented"
+      serialiseToCBOR . asTxMetadata
+
+    deserialiseFromCBOR AsGovernancePoll bs = do
+      metadata <- deserialiseFromCBOR AsTxMetadata bs
+      withNestedMap lbl metadataLabel metadata $ \values ->
+        GovernancePoll
+          -- Question
+          <$> case lookup (TxMetaNumber 0) values of
```

Comment:
For things like `TxMetaNumber 0`, it would be handy to define them as constants in this module.  For

```
metadataKeyQuestion :: TxMetadataValue
metadataKeyQuestion = TxMetaNumber 0
```

Then the above code can change to:

```
<$> case lookup metadataKeyQuestion values of
```

Doing this throughout the file will make it easy to spot when the keys don't line up.

Reply:
Done.